lib/src/Test/Pos/Helpers.hs

```diff
@@ -34,27 +39,52 @@ canonicalJsonTest ::
 canonicalJsonTest =
     identityTest @a $ \x ->
         canonicalJsonRenderAndDecode x .&&. canonicalJsonPrettyAndDecode x
+
+-- | Basically the same as `canonicalJsonTest` but tests a given `Gen a`.
+canonicalJsonTest'
+    :: forall a. (IdTestingRequiredClassesAlmost a, ToAndFromCanonicalJson a)
+    => Gen a
+    -> Spec
+canonicalJsonTest' genA =
```

Comment:
Would `canonicalJsonTestWithGen` or something similar be a more descriptive name?

Reply:
@mhuesch Agree.